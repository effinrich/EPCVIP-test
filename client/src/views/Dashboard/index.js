/** @format */
/* eslint-disable no-console */
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'
import { space } from 'styled-system'

import { fetchUsers } from 'store/users'

const StyledDashboard = styled.div`
  ${space};
`

const Dashboard = () => {
  const dispatch = useDispatch()
  const { users, isLoading } = useSelector(state => state.users)

  useEffect(() => {
    dispatch(fetchUsers())
  }, [dispatch])

  console.log(users, isLoading)

  return <StyledDashboard>Hello</StyledDashboard>
}

export default Dashboard
