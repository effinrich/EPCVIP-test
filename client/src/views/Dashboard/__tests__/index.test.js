import React from 'react'

import { render, screen, fireEvent } from 'utils/testUtils/helperRtl'

import Dashboard from '../index'

// const renderWithRouter = component => {
//   const history = createMemoryHistory()
//   return {
//     ...render(<Router history={history}>{component}</Router>)
//   }
// }

describe('Dashboard View Component', () => {
  it('renders without crashing', () => {
    const { container } = render(<Dashboard />)
    expect(container).toMatchSnapshot()
  })

  it('renders users table', async () => {
    render(<Dashboard />)
    expect(screen.getByRole('table')).toBeInTheDocument()
  })

  it('renders page heading', async () => {
    render(<Dashboard />)
    expect(
      screen.getByRole('heading', { name: /dashboard/i })
    ).toBeInTheDocument()
  })

  it('"Add New User" button click updates route to /add-user', async () => {
    render(<Dashboard />)
    const addNewUserBtn = screen.getByRole('button', { name: /Add New User/i })

    fireEvent.click(addNewUserBtn)

    // expect(
    //   screen.getByRole('heading', { name: /new user/i })
    // ).toBeInTheDocument()
  })
})
