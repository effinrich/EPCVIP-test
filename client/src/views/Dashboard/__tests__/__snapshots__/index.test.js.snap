// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard View Component renders without crashing 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 djCvjv sc-iktFzd zgwlS"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fTeJqU"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 dtaByy"
      >
        Dashboard
      </h1>
      <div
        class="StyledBox-sc-13pk1d4-0 hebQSN"
      >
        <button
          class="StyledButton-sc-323bzc-0 jQIfkU"
          type="button"
        >
          Add New User
        </button>
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 dQYirl"
    >
      <div
        aria-level="1"
        class="sc-fodVxV gbjtWJ rdt_TableHeader"
        role="heading"
      >
        <div
          class="sc-fFubgz fYvZuV"
        />
        <div
          class="sc-bkzZxe iPmElK"
        />
        <div
          class="sc-hBEYos fdFxek"
        >
          <div
            class="sc-bqyKva huJooB"
          />
          <div
            class="sc-kstrdz hfkHbn"
          />
        </div>
      </div>
      <div
        class="sc-hHftDr ddahQf"
      >
        <div
          class="sc-kfzAmx bdkqKp"
        >
          <div
            class="sc-bdfBwQ bmhpxS rdt_Table"
            disabled=""
            role="table"
          >
            <div
              class="sc-bBXqnf dMPErV"
            >
              <div
                style="padding: 24px;"
              >
                There are no records to display
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
